homeassistant/components/zha/alarm_control_panel.py

```diff
@@ -0,0 +1,174 @@
+            cfg_entry, ZHA_ALARM_OPTIONS, CONF_ALARM_MASTER_CODE, "1234"
+        )
+        self._channel.code_required_arm_actions = async_get_zha_config_value(
+            cfg_entry, ZHA_ALARM_OPTIONS, CONF_ALARM_ARM_REQUIRES_CODE, False
+        )
+        self._channel.max_invalid_tries = async_get_zha_config_value(
+            cfg_entry, ZHA_ALARM_OPTIONS, CONF_ALARM_FAILED_TRIES, 3
+        )
+
+    async def async_added_to_hass(self):
+        """Run when about to be added to hass."""
+        await super().async_added_to_hass()
+        self.async_accept_signal(
+            self._channel, SIGNAL_ARMED_STATE_CHANGED, self.async_set_armed_mode
+        )
+        self.async_accept_signal(
+            self._channel, SIGNAL_ALARM_TRIGGERED, self.async_alarm_trigger
+        )
+
+    @callback
+    def async_set_armed_mode(self) -> None:
+        """Set the entity state."""
+        self.async_write_ha_state()
+
+    @property
+    def code_format(self):
+        """Regex for code format or None if no code is required."""
+        return FORMAT_TEXT
+
+    @property
+    def changed_by(self):
+        """Last change triggered by."""
+        return None
+
+    @property
+    def code_arm_required(self):
+        """Whether the code is required for arm actions."""
+        return self._channel.code_required_arm_actions
+
+    async def async_alarm_disarm(self, code=None):
+        """Send disarm command."""
+        self._channel.arm(IasAce.ArmMode.Disarm, code, 0)
+        self.async_write_ha_state()
+
+    async def async_alarm_arm_home(self, code=None):
+        """Send arm home command."""
+        self._channel.arm(IasAce.ArmMode.Arm_Day_Home_Only, code, 0)
+        self.async_write_ha_state()
+
+    async def async_alarm_arm_away(self, code=None):
+        """Send arm away command."""
+        self._channel.arm(IasAce.ArmMode.Arm_All_Zones, code, 0)
+        self.async_write_ha_state()
+
+    async def async_alarm_arm_night(self, code=None):
+        """Send arm night command."""
+        self._channel.arm(IasAce.ArmMode.Arm_Night_Sleep_Only, code, 0)
+        self.async_write_ha_state()
+
+    async def async_alarm_trigger(self, code=None):
+        """Send alarm trigger command."""
+        self.async_write_ha_state()
+
+    @property
+    def supported_features(self) -> int:
+        """Return the list of supported features."""
+        return (
+            SUPPORT_ALARM_ARM_HOME
+            | SUPPORT_ALARM_ARM_AWAY
+            | SUPPORT_ALARM_ARM_NIGHT
+            | SUPPORT_ALARM_TRIGGER
+        )
+
+    @property
+    def state(self):
+        """Return the state of the entity."""
+        return IAS_ACE_STATE_MAP.get(self._channel.armed_state)
+
+    @property
+    def state_attributes(self):
+        """Return the state attributes."""
+        state_attr = {
+            ATTR_CODE_FORMAT: self.code_format,
+            ATTR_CHANGED_BY: self.changed_by,
+            ATTR_CODE_ARM_REQUIRED: self.code_arm_required,
```

Comment:
These attributes are already handled by the base alarm control panel entity class. It's enough to implement the corresponding properties.